File: Two_Pointers/165.Compare-Version-Numbers/165.Compare-Version-Numbers.ts
function compareVersion(version1: string, version2: string): number {
  const v1 = version1.split('.');
  const v2 = version2.split('.');
  const n = Math.max(v1.length, v2.length);

  for (let i = 0; i < n; i++) {
    const patchNo1 = parseInt(v1[i]) || 0;
    const patchNo2 = parseInt(v2[i]) || 0;

    if (patchNo1 > patchNo2) {
      return 1;
    } else if (patchNo1 < patchNo2) {
      return -1;
    }
  }
  return 0;
}

File: Two_Pointers/165.Compare-Version-Numbers/165.Compare-Version-Numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = version1.split('.')
        v2 = version2.split('.')

        if len(v1) > len(v2):
            v2 += [0] * abs(len(v1) - len(v2))
        else:
            v1 += [0] * abs(len(v1) - len(v2))
        
        for i in range(len(v1)):
            if int(v1[i]) > int(v2[i]):
                return 1
            elif int(v1[i]) < int(v2[i]):
                return -1
        
        return 0
